estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,50 @@
+from odoo import api, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property Offer"
+
+    price = fields.Float()
+    status = fields.Selection(
+        string="Type",
+        selection=[("accepted", "Accepted"), ("refused", "Refused")],
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+
+    _sql_constraints = [
+        (
+            "check_positive_price",
+            "CHECK(price>0)",
+            "The offer price should be positive.",
+        ),
+    ]
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self):
+        for record in self:
+            if not record.create_date:
+                record.create_date = fields.Datetime.today()
```

Comment:
You should probably refrain from writing on super fields such as id, create_date, etc...